src/main/java/org/hydev/mcpm/SpigotEntry.java

```diff
@@ -66,4 +72,18 @@ public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command
 
         return true;
     }
+
+    @Override
+    public List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String alias,
+                                      @NotNull String[] args)
+    {
+        if (!command.getName().equalsIgnoreCase("mcpm")) return null;
+
+        if (args.length == 1)
+        {
+            return new ArrayList<>(parser.getRawSubparsers().stream().map(CommandParser::name).toList());
+        }
+
+        return null;
```

Comment:
Should we be returning null instead of empty list? Surely it doesn't matter

Reply:
Spigot supports returning null so it doesn't really matter